packages/m2m-gateway/src/utils/validators/validators.ts

```diff
@@ -30,3 +34,21 @@ export function assertClientVisibilityIsFull(
     );
   }
 }
+
+export function assertActiveConsumerDelegateForEservice(
```

Comment:
These checks are already performed by the delegation process, but I think it makes sense to do them here on the M2M side and fail early in this specific case, since we already retrieve the delegation